### PR TITLE
[5.1] Implemented "Parse multiple repeated request parameters. Use Request::Input to return the array"

### PR DESCRIPTION
**Pull request by FractalizeR**
Implementation of "Parse multiple repeated request parameters. Use Request::Input to return the array " #8434

### CODE REVIEW

File: src/Illuminate/Support/Facades/Input.php [hunk 1]
```diff
@@ -9,16 +9,17 @@ class Input extends Facade
 {
     /**
      * Get an item from the input data.
-     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Facades/Input.php]: Please keep this blank line. It's there for a reason.

File: src/Illuminate/Support/Facades/Input.php [hunk 2]
```diff
@@ -9,16 +9,17 @@ class Input extends Facade
...
      * This method is used for all request verbs (GET, POST, PUT, and DELETE)
      *
      * @param  string  $key
-     * @param  mixed   $default
+     * @param  mixed  $default
+     * @param  null|string  $delimiter
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Facades/Input.php]: string|null not null|string please

File: src/Illuminate/Http/Request.php [hunk 3]
```diff
@@ -328,12 +336,20 @@ public function except($keys)
...
+     *
+     * @return array|string
      */
-    public function query($key = null, $default = null)
+    public function query($key = null, $default = null, $delimiter = null)
     {
-        return $this->retrieveItem('query', $key, $default);
+        $value = $this->retrieveItem('query', $key, $default);
+
+        if (is_string($value) && null !== $delimiter) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Request.php]: please put null on the right of the comparison, not the left

File: src/Illuminate/Http/Request.php [hunk 4]
```diff
@@ -276,14 +276,22 @@ public function all()
...
      */
-    public function input($key = null, $default = null)
+    public function input($key = null, $default = null, $delimiter = null)
     {
         $input = $this->getInputSource()->all() + $this->query->all();
 
-        return array_get($input, $key, $default);
+        $value = array_get($input, $key, $default);
+
+        if (is_string($value) && null !== $delimiter) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Request.php]: please put null on the right of the comparison, not the left

**Comment by GrahamCampbell** [comment]
:-1:

**Comment by GrahamCampbell** [comment]
this is inconsistant

**Comment by RomainLanz** [comment]
:-1: 

As @GrahamCampbell said on the issue...
> Repeated parameters make no sense and should be avoided without exception.

**Comment by FractalizeR** [comment]
Well, you are probably right. It is just I saw @taylorotwell commenting this positively and decided to make implementation: https://github.com/laravel/framework/issues/8434#issuecomment-97110530

May be, I misunderstood him.

**Comment by GrahamCampbell** [comment]
This is inconsistant, because the data from this doesn;t appear in "all", or "except", or does it?

**Comment by GrahamCampbell** [comment]
Please add tests for that.

**Comment by franzliedke** [comment]
Well, this is a parsing helper (I still think it should be a separate method, but this seems to be what Taylor wanted), so I'd argue it's fine to appear in its original form in "all" etc.

**Comment by FractalizeR** [comment]
I will add some tests tonight.

**Comment by GrahamCampbell** [comment]
> I will add some tests tonight.

Thank you. With some decent test coverage, I'm not massively against this anymore. :)

**Comment by garygreen** [comment]
I'm -1 to this, it should be a helper if anything. Also chances are if you need to do this your probably not using array based HTTP parameters and should be (cough select2)

**Comment by taylorotwell** [comment]
Sounds like reaction is mixed so I'm going to hold because we're quite close to release and it's fairly easily implemented as a helper.

**Pull request closed without merging**